pkg/services/mattermost.go

```diff
@@ -37,9 +38,13 @@ func (n *MattermostNotification) GetTemplater(name string, f texttemplate.FuncMa
 }
 
 type MattermostOptions struct {
-	ApiURL             string `json:"apiURL"`
-	Token              string `json:"token"`
-	InsecureSkipVerify bool   `json:"insecureSkipVerify"`
+	ApiURL              string        `json:"apiURL"`
+	Token               string        `json:"token"`
+	InsecureSkipVerify  bool          `json:"insecureSkipVerify"`
+	MaxIdleConns        int           `json:"maxIdleConns"`
+	MaxIdleConnsPerHost int           `json:"maxIdleConnsPerHost"`
+	MaxConnsPerHost     int           `json:"maxConnsPerHost"`
+	IdleConnTimeout     time.Duration `json:"idleConnTimeout"`
```

Comment:
We can't use time.Duration for parsing time values - we will need to use a string and then parse the actual `time.Duration` by doing something like this 

```golang
	idleConnTimeout, err := time.ParseDuration(s.opts.IdleConnTimeout)
	if err != nil {
		return errors.Wrap(err, "failed to parse idle connection timeout")
	}
```

and then passing in `idleConnTimeout` to the `NewTransport` function.